merge_sort.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+import time
+import random
+
+
+def timed_func(func):
+    """Decorator for timing our traversal methods."""
+    def timed(*args, **kwargs):
+        start = time.time()
+        result = func(*args, **kwargs)
+        elapsed = time.time() - start
+        print "time expired: %s" % elapsed
+        return result
+    return timed
+
+
+@timed_func
+def time_merge_sort(sequence):
+    return merge_sort(sequence)
+
+
+def merge_sort(sequence):
+    """Mergesort."""
+    for ind, num in enumerate(sequence):
+        # base case
+        if len(sequence) <= 1:
+            return sequence
```

Comment:
shouldn't this test really be outside the loop you've built here?  There's no reason to even try to loop if the list is only one item long or less, no?